src/platform/packages/shared/kbn-scout/src/playwright/page_objects/date_picker.ts

```diff
@@ -41,6 +51,11 @@ export class DatePicker {
     }
   }
 
+  async clickSubmitButton() {
+    await this.page.testSubj.click('querySubmitButton');
+    await this.page.waitForLoadingIndicatorHidden();
```

Comment:
I think we already discussed that waiting for "global loading indicator" won't be a reliable solution here: it will still be loading whenever any incoming request reaches the client.

Instead, let's wait for a specific condition case by case: chart, doc table, etc.